lib/src/geometry/collision_detection.dart

```diff
@@ -3,11 +3,32 @@ import '../components/mixins/collidable.dart';
 
 /// Check whether any [Collidable] in [collidables] collide with each other and
 /// call their onCollision methods accordingly.
-void collisionDetection(List<Collidable> collidables) {
-  for (var x = 0; x < collidables.length; x++) {
-    final collidableX = collidables[x];
-    for (var y = x + 1; y < collidables.length; y++) {
-      final collidableY = collidables[y];
+void collisionDetection(
+  List<Collidable> collidables,
+) {
+  final activeCollisionList = <Collidable>[];
```

Comment:
I think all the variables could drop the "List" part and made to plural, so this would be `activeCollidables` for example.

Reply:
And they are collidables, not collisions

Reply:
OK I'll make the changes right away